testing/src/scenario/state.py

```diff
@@ -1046,12 +1052,36 @@ def __post_init__(self):
     def _render_services(self):
         # copied over from ops.testing._TestingPebbleClient._render_services()
         services: dict[str, pebble.Service] = {}
-        for key in sorted(self.layers.keys()):
-            layer = self.layers[key]
+        for layer in self.layers.values():
             for name, service in layer.services.items():
-                services[name] = service
+                if name in services and service.override == "merge":
+                    services[name]._merge(service)
+                else:
+                    services[name] = service
         return services
 
+    def _render_checks(self) -> Dict[str, pebble.Check]:
+        # copied over from ops.testing._TestingPebbleClient._render_checks()
```

Comment:
Too tricky to reuse the same/similar code from `_private/harness.py`?

Reply:
"copied over" is a little imprecise (I borrowed the wording from the `render_services` above). One uses `self.checks` (and so on) and the other `self._checks` (and so on). I could pass in the attribute name, I guess.

I'm reluctant to make Scenario *more* dependant on Harness, since the aim is that eventually Harness breaks off into `ops-harness` and `ops-scenario` depends only on `ops` and not that new package. On the other hand, there's plenty of dependency at the moment, so it wouldn't be making things significantly worse.

I'll have a little play with it.

Reply:
Ah got it -- in that case, I think it's fine as is. "A little copying is better than a little dependency" :-)